sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -194,6 +194,46 @@ public String getServiceBusResourceName() {
         return defer(message, null);
     }
 
+    /**
+     * Asynchronously renews the lock on the message specified by the lock token. The lock will be renewed based on the
+     * setting specified on the entity. When a message is received in {@link ReceiveMode#PEEK_LOCK} mode, the message is
+     * locked on the server for this receiver instance for a duration as specified during the Queue creation
+     * (LockDuration). If processing of the message requires longer than this duration, the lock needs to be renewed.
+     * For each renewal, the lock is reset to the entity's LockDuration value.
+     *
+     * @param messageLock The {@link UUID} value of the message lock to renew.
+     *
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Instant> renewMessageLock(UUID messageLock) {
+        return connectionProcessor
+            .flatMap(connection -> connection.getManagementNode(entityPath, entityType))
+            .flatMap(serviceBusManagementNode -> serviceBusManagementNode
+                .renewMessageLock(messageLock));
+    }
+
+    /**
+     * Asynchronously renews the lock on the specified message. The lock will be renewed based on the
+     * setting specified on the entity. When a message is received in {@link ReceiveMode#PEEK_LOCK} mode, the message is
+     * locked on the server for this receiver instance for a duration as specified during the Queue creation
+     * (LockDuration). If processing of the message requires longer than this duration, the lock needs to be renewed.
+     * For each renewal, the lock is reset to the entity's LockDuration value.
+     *
+     * @param receivedMessage to be used to renew.
+     *
+     * @return The {@link Mono} the finishes this operation on service bus resource.
+     */
+    public Mono<Instant> renewMessageLock(ServiceBusReceivedMessage receivedMessage) {
+        return connectionProcessor
+            .flatMap(connection -> connection.getManagementNode(entityPath, entityType))
+            .flatMap(serviceBusManagementNode -> serviceBusManagementNode
+                .renewMessageLock(receivedMessage.getLockToken())
+                .map(instant -> {
+                    receivedMessage.setLockedUntil(instant);
```

Comment:
is this what they do in track 1? update the actual parameter object itself? Since we want to make ServicebusReceivedMessage immutable, id expect a clone to be created.

Reply:
Track 1: They update same object in this way.